x-pack/legacy/plugins/infra/common/inventory_models/aws_ec2/metrics/tsvb/aws_ec2_cpu_utilization.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TSVBMetricModelCreator, TSVBMetricModel } from '../../../types';
+
+export const awsEC2CpuUtilization: TSVBMetricModelCreator = (
```

Comment:
What about creating some util funcs for these modelCreators. For example:

```ts
const createAWSEC2Model = (indexPattern: string, timeField: string, series: any) => {
    return {
      requires: ['aws.ec2'],
      .....
      series
    }
}
```


```ts
createAWSEC2Model([
    {
      id: 'write',
      split_mode: 'everything',
      metrics: [
        {
          field: 'aws.ec2.diskio.write.bytes_per_sec',
          id: 'avg-write',
          type: 'avg',
        },
      ],
    },
    {
      id: 'read',
      split_mode: 'everything',
      metrics: [
        {
          field: 'aws.ec2.diskio.read.bytes_per_sec',
          id: 'avg-read',
          type: 'avg',
        },
        {
          id: 'calculation-rate',
          type: 'calculation',
          variables: [{ id: 'rate-var', name: 'rate', field: 'avg-read' }],
          script: 'params.rate * -1',
        },
      ],
    },
  ]);
```

Doing that could save about 70 lines of code in this PR alone.

Reply:
That would work... I was also thinking for all the basic ones where it's just an avg/max/min we could just create a function that takes the field name. I'm sure that would reduce things down some too.